src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -557,7 +557,11 @@ def delete_ag_url_path_map_rule(resource_group_name, application_gateway_name, u
         resource_group_name, application_gateway_name, ag)
 
 def set_ag_waf_config(resource_group_name, application_gateway_name, enabled,
-                      firewall_mode=ApplicationGatewayFirewallMode.detection.value, no_wait=False):
+                      firewall_mode=None, no_wait=False):
```

Comment:
To hide this on unsupported API versions, you can use the new `VersionConstraint` class in `azure.cli.core`:

```Python
with VersionConstraint(ResourceType.MGMT_NETWORK, min_api='...') as c:
   c.register_cli_argument('application-gateway', 'firewall_mode', ...)
```

This would be the equivalent of:

```Python
if supported_api_version(ResourceType.MGMT_NETWORK, min_api='...'):
   register_cli_argument('application-gateway', 'firewall_mode',...)
else:
   register_cli_argument('application-gateway', 'firewall_mode', ignore_type)
```